### PR TITLE
Fix unable to delete an image from the media-library

### LINKED ISSUE #11939: [v4] Cannot delete an image from the media-library
### Describe the bug

Uploaded Image cannot be deleted from admin panel, request failed with status code 500

### Steps to reproduce the behavior

1. curl https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png -o lenna.png
2. curl -X 'POST'   'http://localhost:1337/api/upload' -F 'files=@lenna.png'
3. go to strapi admin-> media-library
4. select lenna.png
5. click on button "Delete"
6. click on "Confirm" on the popup

### Expected behavior

The uploaded image is deleted from the media-library

### Screenshots

Schermata da 2021-12-17 12-52-20

### Code snippets

This is what the log says for the 500 error:

```
TypeError: Cannot read properties of null (reading 'id')
    at findEntityAndCheckPermissions (/srv/app/node_modules/@strapi/plugin-upload/server/controllers/admin-api.js:197:86)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.destroy (/srv/app/node_modules/@strapi/plugin-upload/server/controllers/admin-api.js:68:26)
    at async returnBodyMiddleware (/srv/app/node_modules/@strapi/strapi/lib/services/server/compose-endpoint.js:52:18)
    at async policiesMiddleware (/srv/app/node_modules/@strapi/strapi/lib/services/server/policy.js:24:5)
    at async /srv/app/node_modules/@strapi/strapi/lib/middlewares/body.js:24:7
    at async /srv/app/node_modules/@strapi/strapi/lib/middlewares/logger.js:22:5
    at async /srv/app/node_modules/@strapi/strapi/lib/middlewares/powered-by.js:16:5
    at async cors (/srv/app/node_modules/@koa/cors/index.js:95:16)
    at async /srv/app/node_modules/@strapi/strapi/lib/middlewares/errors.js:13:7
    at async session (/srv/app/node_modules/koa-session/index.js:41:7)
    at async /srv/app/node_modules/@strapi/strapi/lib/services/metrics/middleware.js:29:5
If applicable, add code samples to help explain your problem.
```
### System

- Node.js version: 16.13.0
- NPM version: 8.1.0
- Strapi version:v4
- Database: postgres
- Operating system: debian

### PR DESCRIPTION
**Pull request by iicdii**
Signed-off-by: harimkims <harimkims@gmail.com>

### What does it do?

- Avoid direct access to created by attribute in file, Use [lodash](https://lodash.com/) get instead.

### Why is it needed?

Because some uploaded files doesn't have `createdBy` attribute - for example when you upload file as a public Role - It throws an error when it does not have author.

### How to test it?

1. Give upload permission to the `Public` role.
2. `curl https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png -o lenna.png`
3. `curl -X 'POST' 'http://localhost:1337/api/upload' -F 'files=@lenna.png'`
4. Go to strapi admin-> media-library
5. Select lenna.png
6. Click on button "Delete"
7. Click on "Confirm" on the popup

### Related issue(s)/PR(s)

Closes #11939

### CODE REVIEW

File: packages/core/upload/server/controllers/admin-api.js [hunk 1]
```diff
@@ -194,7 +194,10 @@ const findEntityAndCheckPermissions = async (ability, action, model, id) => {
 
   const pm = strapi.admin.services.permission.createPermissionsManager({ ability, action, model });
 
-  const author = await strapi.admin.services.user.findOne(file[CREATED_BY_ATTRIBUTE].id, ['roles']);
+  const author = await strapi.admin.services.user.findOne(
```
Comment (alexandrebodin) [inline packages/core/upload/server/controllers/admin-api.js:197]: that would rather deserve to be a conditional from the looks of it. if their is no created_by you don't want to call findOne(' ') with an empty string

Comment (iicdii) [reply packages/core/upload/server/controllers/admin-api.js:197]: I agree with you.
```suggestion
  const creatorId = _.get(file, [CREATED_BY_ATTRIBUTE, 'id'], '');
  const author = creatorId ? await strapi.admin.services.user.findOne(creatorId, ['roles']) : null;
```
What do you think?

Comment (alexandrebodin) [reply packages/core/upload/server/controllers/admin-api.js:197]: lgtm

Comment (alexandrebodin) [reply packages/core/upload/server/controllers/admin-api.js:197]: you can leave creatorId undefiend instead of empty string though by default

Comment (iicdii) [reply packages/core/upload/server/controllers/admin-api.js]: Done :)

**Review by alexandrebodin — APPROVED** [review]
Looks good to me 👍

**Pull request merged**